### ISSUE TITLE
False positive PH2089: Anonymous object initialization mistaken for Assignment in condition

### ISSUE THREAD

**Issue by jafaber** [philips-software/roslyn-analyzers#776]
Simplified minimal example.

```cs
using System;

int blah1 = 10;

if(Log(new{ BLAH = blah1 })) // <---  PH2089: Assignment in condition.
{
}

bool Log(object x) {
    Console.WriteLine(x);
    return true;
}
```
